central/audit/audit.go

```diff
@@ -152,6 +153,15 @@ func (a *audit) UnaryServerInterceptor() func(ctx context.Context, req interface
 	}
 }
 
+func (a *audit) HTTPInterceptor(handler http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		statusTrackingWriter := httputil.NewStatusTrackingWriter(w)
+		handler.ServeHTTP(statusTrackingWriter, r)
+
+		go a.SendAdhocAuditMessage(r.Context(), r, r.RequestURI, interceptor.AuthStatus{}, statusTrackingWriter.GetStatusCodeError())
```

Comment:
We could also set`authError` here by comparing status code to 401/403

Reply:
Currently, this handler is called after auth. Failing to auth breaks the handler chain and nothing is logged. I have added a comment to the ticket related to this.

I would not add any additional complexity here. In theory, we should be able to call `interceptor.GetAuthErrorFromContext(r.Context())` - but, probably request context is not enriched properly and, there is no auth information. When I call that function, I always get: `"status":"AUTH_FAILED","statusReason":"authentication status is always required"`

I have changed function name to `PostAuthHTTPInterceptor` to clearly signal the intention.